### PR TITLE
Running doctests on Windows

### LINKED ISSUE #322: Run Windows tests on `nightly` toolchain to allow x-compiled doctests
Doctests are not executed during crosscompilation, which is the case of all Windows builds (`x64 msvc` -> `x64 gnu` is also cross-compilation). However, there is a feature available in nightly builds [`-Zdoctest-xcompile`](https://github.com/rust-lang/cargo/issues/7040), which runs cross-compiled doc tests.
As the majority of our tests are doc tests, it seems important for the next several versions of R (until 32 bit support is dropped).

Proposed changes:
1. Choose here
https://github.com/extendr/extendr/blob/072945f3ee87ea93cdea3652a59ee85e572880a1/.github/workflows/test.yml#L277
`nightly` pipeline for Windows and add `i686-pc-windows-gnu` target
https://github.com/extendr/extendr/blob/072945f3ee87ea93cdea3652a59ee85e572880a1/.github/workflows/test.yml#L285-L288
2. Fix environment setup
https://github.com/extendr/extendr/blob/072945f3ee87ea93cdea3652a59ee85e572880a1/.github/workflows/test.yml#L353-L362
3. Conditionally add `-Zdoctest-xcompile` if running on Windows and on `nightly` toolchain
https://github.com/extendr/extendr/blob/072945f3ee87ea93cdea3652a59ee85e572880a1/.github/workflows/test.yml#L434-L440

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
Closes #322

### CODE REVIEW

**Comment by Ilia-Kosenkov** [comment]
I have also fully disabled NA tests referencing #321 until the floating point issue is resolved.

**Comment by Ilia-Kosenkov** [comment]
Now we have some sort of a benchmark for i686

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -285,13 +285,12 @@ jobs:
         config:
           - {os: ubuntu-20.04,   r: 'devel', rust-version: 'stable', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}
           - {os: macOS-latest,   r: 'devel', rust-version: 'stable'}
-          - {os: windows-latest, r: 'devel', rust-version: 'stable', rust-targets: ['x86_64-pc-windows-gnu']}
+          - {os: windows-latest, r: 'release', rust-version: 'nightly-msvc', rust-targets: ['x86_64-pc-windows-gnu', 'i686-pc-windows-gnu'], extra-args: ['-Zdoctest-xcompile']}
```
Comment (yutannihilation) [inline .github/workflows/test.yml:288]: Can you add some comment that we'll change this back to r-devel after R 4.2 is released?

**Review by yutannihilation — APPROVED** [review]
Sorry for the delay, looks good to me!

File: .github/workflows/test.yml (hunk 1 above)
Comment (Ilia-Kosenkov) [reply .github/workflows/test.yml:289]: Done

Comment (yutannihilation) [reply .github/workflows/test.yml:289]: Thanks!

**Pull request merged**